src/NServiceBus.Transport.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -506,6 +506,7 @@ public void Dispose()
         {
             circuitBreaker?.Dispose();
             semaphore?.Dispose();
+            messageProcessing?.Cancel();
```

Comment:
```suggestion
            // This makes sure that if the stop token hasn't been canceled the processing source is canceled
            // so that any possible reconnect attempt has the possibility to gracefully stop too.
            messageProcessing?.Cancel();
```